sw/device/tests/i2c_host_compass_test.c

```diff
@@ -69,12 +69,14 @@ static status_t take_measurement(void) {
   // Poll until status indicates measurement was taken.
   uint8_t status = 0;
   uint8_t status_reg = kStatusReg;
-  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &status_reg, true));
   do {
```

Comment:
Should there be a timeout for the whole loop? For example what if the device does not report a measurement done after some time? I guess this would hit the test timeout but I feel like handling this explicitely would help make failures more understandable (test timeouts are super informative).
Feel free to ignore this if it's overkill.

Reply:
The test will timeout after 1 minute and there's log messages enough to indicate where it blocked. So I think we are safe.